### PR TITLE
Uphold Integration

### PR DESCRIPTION
**Pull request by sambarboza**
# Description
- Module that integrates Uphold and allows user to buy/sell Dash using an Uphold account and create a new one if necessary.
- All interactions but transferring Dash from Uphold to the Wallet happens inside a WebView due to Uphold requirements.

# Evidence / Screens

### CODE REVIEW

**Comment by juankstylez** [comment]
excellent feature! support for !!

**Comment by HashEngineering** [comment]
the uphold integration module has the package com.dash.wallet.* while the common module has org.dash.wallet.*  

Is there a reason for this?

**Comment by HashEngineering** [comment]
This fails CI with these errors:

```
Could not resolve all files for configuration ':common:debugCompileClasspath'.
> Could not find support-vector-drawable.aar (com.android.support:support-vector-drawable:27.1.1).
  Searched in the following locations:
      https://jcenter.bintray.com/com/android/support/support-vector-drawable/27.1.1/support-vector-drawable-27.1.1.aar
> Could not find livedata-core.aar (android.arch.lifecycle:livedata-core:1.1.0).
  Searched in the following locations:
      https://jcenter.bintray.com/android/arch/lifecycle/livedata-core/1.1.0/livedata-core-1.1.0.aar
> Could not find viewmodel.aar (android.arch.lifecycle:viewmodel:1.1.0).
  Searched in the following locations:
      https://jcenter.bintray.com/android/arch/lifecycle/viewmodel/1.1.0/viewmodel-1.1.0.aar
> Could not find runtime.aar (android.arch.core:runtime:1.1.0).
  Searched in the following locations:
      https://jcenter.bintray.com/android/arch/core/runtime/1.1.0/runtime-1.1.0.aar
```

**Comment by HashEngineering** [comment]
if we put the Google Repository first:  

Then I get this set of errors using `gradlew build`:

These errors seem to be from missing resources in `common` that are found in `wallet`

```
  Copying resources from program directory [C:\Users\HashEngineering\Documents\HashEngineering\Dash\dash-wallet\common\build\intermediates\classes\release] (filtered)
  Copying resources from program jar [C:\Users\HashEngineering\Documents\HashEngineering\Dash\dash-wallet\common\build\intermediates\transforms\mergeJavaRes\release\0.jar] (filtered)
Printing classes to [C:\Users\HashEngineering\Documents\HashEngineering\Dash\dash-wallet\common\build\outputs\mapping\release\dump.txt]...
C:\Users\HashEngineering\Documents\HashEngineering\Dash\dash-wallet\common\src\main\res\values\values.xml:10:2-16:17: AAPT: error: resource string/preferences_precision_labels_8 (aka org.dash.wallet.common:
string/preferences_precision_labels_8) not found.

C:\Users\HashEngineering\Documents\HashEngineering\Dash\dash-wallet\common\src\main\res\values\values.xml:10:2-16:17: AAPT: error: resource string/preferences_precision_labels_6 (aka org.dash.wallet.common:
string/preferences_precision_labels_6) not found.

C:\Users\HashEngineering\Documents\HashEngineering\Dash\dash-wallet\common\src\main\res\values\values.xml:10:2-16:17: AAPT: error: resource string/preferences_precision_labels_4 (aka org.dash.wallet.common:
string/preferences_precision_labels_4) not found.

C:\Users\HashEngineering\Documents\HashEngineering\Dash\dash-wallet\common\src\main\res\values\values.xml:10:2-16:17: AAPT: error: resource string/preferences_precision_labels_2_3 (aka org.dash.wallet.commo
n:string/preferences_precision_labels_2_3) not found.

C:\Users\HashEngineering\Documents\HashEngineering\Dash\dash-wallet\common\src\main\res\values\values.xml:10:2-16:17: AAPT: error: resource string/preferences_precision_labels_0_6 (aka org.dash.wallet.commo
n:string/preferences_precision_labels_0_6) not found.

C:\Users\HashEngineering\Documents\HashEngineering\Dash\dash-wallet\common\src\main\res\values\styles.xml:307:5-312:13: AAPT: error: resource id/transaction_row_address_container (aka org.dash.wallet.common
:id/transaction_row_address_container) not found.

C:\Users\HashEngineering\Documents\HashEngineering\Dash\dash-wallet\common\src\main\res\values\styles.xml:307:5-312:13: AAPT: error: resource id/transaction_row_address_container (aka org.dash.wallet.common
:id/transaction_row_address_container) not found.

C:\Users\HashEngineering\Documents\HashEngineering\Dash\dash-wallet\common\src\main\res\values\styles.xml:307:5-312:13: AAPT: error: resource id/transaction_row_value (aka org.dash.wallet.common:id/transact
ion_row_value) not found.

C:\Users\HashEngineering\Documents\HashEngineering\Dash\dash-wallet\common\src\main\res\values\styles.xml:307:5-312:13: AAPT: error: resource id/transaction_row_value (aka org.dash.wallet.common:id/transact
ion_row_value) not found.

C:\Users\HashEngineering\Documents\HashEngineering\Dash\dash-wallet\common\src\main\res\values\styles.xml:307:5-312:13: AAPT: error: resource id/transaction_row_address_container (aka org.dash.wallet.common
:id/transaction_row_address_container) not found.

C:\Users\HashEngineering\Documents\HashEngineering\Dash\dash-wallet\common\src\main\res\values\styles.xml:167:5-169:13: AAPT: error: resource color/inverted_button (aka org.dash.wallet.common:color/inverted
_button) not found.

C:\Users\HashEngineering\Documents\HashEngineering\Dash\dash-wallet\common\src\main\res\values\styles.xml:167:5-169:13: AAPT: error: resource drawable/inverted_button (aka org.dash.wallet.common:drawable/in
verted_button) not found.

error: failed linking references.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':common:verifyReleaseResources'.
> com.android.ide.common.process.ProcessException: Failed to execute aapt
```

File: common/src/main/java/org/dash/wallet/common/Constants.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    public static final char CHAR_HAIR_SPACE = '\u200a';
+    public static final char CHAR_THIN_SPACE = '\u2009';
+    public static final char CHAR_ALMOST_EQUAL_TO = '\u2248';
+    public static final char CHAR_CHECKMARK = '\u2713';
+    public static final char CURRENCY_PLUS_SIGN = '\uff0b';
+    public static final char CURRENCY_MINUS_SIGN = '\uff0d';
+    public static final String PREFIX_ALMOST_EQUAL_TO = Character.toString(CHAR_ALMOST_EQUAL_TO) + CHAR_THIN_SPACE;
+
+    //This is used for UI only
+    public static final boolean TEST = BuildConfig.DEBUG;
```
[the remark this answers is not in the document]
Comment (HashEngineering) [reply common/src/main/java/org/dash/wallet/common/Constants.java]: This matter will be resolved in a different PR.

**Pull request merged**